compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt

```diff
@@ -204,6 +210,21 @@ class SkikoComposeUiTest @InternalTestApi constructor(
         }
     }
 
+    private inline fun <R> renderingAtFrameRate(block: () -> R): R {
+        val scope = CoroutineScope(coroutineContext)
+        return try {
+            scope.launch {
+                while (isActive) {
+                    delay(FRAME_DELAY_MILLIS)
+                    render(mainClock.currentTime)
```

Comment:
Hmm, I think we should remove advancing the clock in `renderNextFrame` instead. It's used in 3 places:
- `waitForIdle`, where `isIdle` already advances the clock.
- `awaitForIdle`, likewise.
- `waitUntil`, where we should add advancing the clock.

Reply:
> should remove advancing the clock in renderNextFrame instead

Makes sense, but should we call `render` only if `mainClock.autoAdvance = false`?

Reply:
> Makes sense, but should we call render only if mainClock.autoAdvance = false?

Not sure I understand. Why would we not want to render if `mainClock.autoAdvance` is `true`?

Reply:
My thought is that now only the render cycle should own and have responsibility for calling `render`.

Calling it from 2 places causes:
- race conditions (though `runOnUiThread` will prevent that)
- calling it 2 times: first explicitly via `render` and second via a request `advanceTimeByFrame`, which already implies that it should definitely `render` the frame.

Reply:
> the render cycle should own and have responsibility for calling render.

But `mainClock.autoAdvance = false` slightly breaks this logic, because it says that we should still call `layout` and `draw`, which we call in `render`

Reply:
Ah, I see. I'll look into it.